### PR TITLE
Query: Fixes incorrect order by query when the field is an object or array

### PR DESCRIPTION
**Pull request by kristynhamasaki**
# Pull Request Template

## Description

Currently, V3 SDK hashing of object/arrays and comparisons of the hashes does not match backend behavior. This PR updates them to match. 

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/DistinctHashBaselineTests.ElementsHash.xml [hunk 1]
```diff
@@ -52,7 +52,7 @@
       <CosmosElement><![CDATA[[null,false,true,1234,"asdfasdfasdfasdfasdfasdf"]]]></CosmosElement>
     </Input>
     <Output>
-      <Hash><![CDATA[25-69-E9-CB-22-6C-58-6E-C0-0B-1F-27-79-F6-9A-16]]></Hash>
+      <Hash><![CDATA[A9-40-79-16-DE-27-F3-16-CC-5A-8E-4E-B9-AF-D5-E6]]></Hash>
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/BaselineTest/TestBaseline/DistinctHashBaselineTests.ElementsHash.xml:55]: Can a test be added to catch this scenario? Just looking at hash code make it easy to miss issues.

**Review by j82w — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/BinaryComparer.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Compare UInt128's by least significant byte to most significant byte.
+    /// </summary>
+    internal sealed class BinaryComparer : IComparer<UInt128>
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/BinaryComparer.cs:12]: >BinaryComparer [](http://example.com/codeflow?start=26&length=14)

UInt128BinaryComparer (also the file name)

File: Microsoft.Azure.Cosmos/src/BinaryComparer.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        public static readonly BinaryComparer Singleton = new BinaryComparer();
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/BinaryComparer.cs:14]: >public static readonly BinaryComparer Singleton = new BinaryComparer(); [](http://example.com/codeflow?start=8&length=71)

please add a private constructor as well

File: Microsoft.Azure.Cosmos/src/BinaryComparer.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        public int Compare(UInt128 x, UInt128 y)
+        {
+            if (x.GetLow() != y.GetLow())
+            {
+                if (this.ReverseBytes(x.GetLow()) < this.ReverseBytes(y.GetLow()))
+                {
+                    return -1;
+                }
+                else
```
Comment (sboshra) [inline Microsoft.Azure.Cosmos/src/BinaryComparer.cs:24]: >else [](http://example.com/codeflow?start=16&length=4)

no need for else here

**Review by sboshra — DISMISSED** [review]
:shipit:

File: Microsoft.Azure.Cosmos/src/BinaryComparer.cs (hunk 2 above)
Comment (kristynhamasaki) [reply Microsoft.Azure.Cosmos/src/BinaryComparer.cs]: changed name

File: Microsoft.Azure.Cosmos/src/BinaryComparer.cs (hunk 4 above)
Comment (kristynhamasaki) [reply Microsoft.Azure.Cosmos/src/BinaryComparer.cs]: removed

File: Microsoft.Azure.Cosmos/src/BinaryComparer.cs (hunk 3 above)
Comment (kristynhamasaki) [reply Microsoft.Azure.Cosmos/src/BinaryComparer.cs]: done

**Review by sboshra — APPROVED** [review]

**Pull request merged**